serf/serf.go

```diff
@@ -741,6 +741,19 @@ func (s *Serf) Members() []Member {
 	return members
 }
 
+// CheckAndRemoveFailedNode checks if a node exists and then forcibly removes
+// the node via RemoveFailedNode. If the node does not exist in the cluster, it
+// will return exists as false and will not try to remove the node
+func (s *Serf) CheckAndRemoveFailedNode(node string) (exists bool, err error) {
+	// Check to make sure that the node exists in the cluster
+	_, ok := s.members[node]
+	if !ok {
+		return false, nil
```

Comment:
I am not sure if it is ok to return here or if we should check for membership and keep going with the removal even if it's not in the member list. I'm not sure if `s.members` is considered a complete list or if it _might_ not be 100% up to date.